lib/utils.ex

```diff
@@ -15,6 +15,55 @@ defmodule BlockBox.Utils do
     end)
   end
 
+  def convert_initial_opts(opts) do
+    select_key =
+      cond do
+        Keyword.has_key?(opts, :initial_option) -> :initial_option
+        Keyword.has_key?(opts, :initial_options) -> :initial_options
+        true -> nil
+      end
+
+    case select_key do
+      nil ->
+        opts
+
+      select_key ->
+        opts
+        |> Keyword.update(select_key, nil, fn
+          x when is_list(x) -> x
+          x -> [x]
+        end)
+        |> convert_initial_opts_helper(select_key)
+    end
+  end
+
+  defp convert_initial_opts_helper(opts, put_opt) do
+    items =
+      case Keyword.get(opts, put_opt) do
+        [val | _tail] = initials when is_integer(val) ->
+          options = Keyword.get(opts, :options)
+          Enum.map(initials, fn index -> options |> Enum.at(index) end)
+
+        [val | _tail] = initials when is_tuple(val) ->
+          option_groups = Keyword.get(opts, :option_groups)
+
+          Enum.map(initials, fn {grp_ind, opt_ind} ->
```

Comment:
it's actually `num_indices * (m+n)` worst case.
I'll stick with this one because the implementation is simpler and faster in the worst case than a version that narrows down the groups first. Also, it doesn't matter that much because the lists we're expecting are gonna be max a dozen items.